packages/core/src/checkPackage.ts

```diff
@@ -47,7 +57,14 @@ export async function checkPackage(pkg: Package, options?: CheckPackageOptions):
   }
 
   const hosts = createCompilerHosts(pkg);
-  const entrypointResolutions = getEntrypointInfo(packageName, pkg, hosts, options);
+  let files: string[] | undefined = undefined;
+  if (options?.entrypointsLegacy && dir) {
+    const path = join(process.cwd(), dir);
+    const tree = await new Arborist().loadActual({ path });
+    const extensions = new Set([".js", ".jsx", ".ts", ".tsx", ".mjs", ".cjs", ".mts", ".cts"]);
+    files = (await packlist(tree, { path })).filter(x => !x.endsWith('.d.ts') && extensions.has(extname(x))).map(x => `./${x}`);
+  }
```

Comment:
this logic really shouldn't be here - it should be inside `getEntrypointInfo`, but it seemed like making it (and getEntrypoints) an async function was a larger change than i should be making.

happy to do that, tho, if you prefer.